conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py

```diff
@@ -68,17 +68,22 @@ def run(known_args, pipeline_args):
   with beam.Pipeline(options=pipeline_options) as p:
     pcoll = (p | ReadFromText(known_args.input)
                    | beam.ParDo(WriteDecompressedFile())
-                   | beam.io.Write(bigquery.BigQuerySink(known_args.table, schema=known_args.schema)))
+                   | beam.io.Write(bigquery.BigQuerySink(known_args.table, schema=known_args.schema, validate = True)))
 
 def truncate_content(s):
     dic = json.loads(s) 
     dic['truncated'] = False
+    dic['no_records'] = 1
+    dic['record_index'] = 0
     if sys.getsizeof(s) > THERESHOLD:
        l = len(dic['text'])
        dic['truncated'] = True
+       dic['no_records'] = 2
        dic1 = copy.deepcopy(dic)
        dic1['text'] = dic1['text'][:l/2]
+       dic1['record_index'] = 0
```

Comment:
Would be better to decide on a length (maybe 5MB) and break it up into as many chunks as needed, rather than just divide by two. E.g. What if you get a 30mb chunk?

Reply:
I broke down the records given the size.
But didn't find a way to read specific lines from a text file in the api, was looking at this documentation(ReadFromText part): https://beam.apache.org/documentation/sdks/pydoc/0.6.0/_modules/apache_beam/io/textio.html